status/views.py

```diff
@@ -0,0 +1,158 @@
+  "up," DOWN, or NO_CONFIG for the "status" key.
+"""
+
+from datetime import datetime
+import logging
+import signal
+from pprint import pformat
+
+from django.conf import settings
+from django.http import JsonResponse, Http404
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from kombu.utils.url import _parse_url as parse_redis_url
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+UP = "up"
+DOWN = "down"
+NO_CONFIG = "no config found"
+HTTP_OK = 200
+SERVICE_UNAVAILABLE = 503
+
+
+# A timeout context manager.
+# Found:
+# http://stackoverflow.com/questions/8464391/what-should-i-do-if-socket-setdefaulttimeout-is-not-working
+class Timeout():
+    """Timeout class using ALARM signal"""
+    class Timeout(Exception):
+        """Custom exception."""
+    def __init__(self, sec):
+        self.sec = sec
+
+    def __enter__(self):
+        signal.signal(signal.SIGALRM, self.raise_timeout)
+        signal.alarm(self.sec)
+
+    def __exit__(self, *args):
+        signal.alarm(0)  # disable alarm
+
+    # pylint: disable=no-self-use,unused-argument
+    def raise_timeout(self, *args):
+        """Raise custom exception."""
+        raise Timeout.Timeout()
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    log.debug("entered get_pg_info")
+    try:
+        conf = settings.DATABASES['default']
+        database = conf["NAME"]
+        user = conf["USER"]
+        host = conf["HOST"]
+        port = conf["PORT"]
+        password = conf["PASSWORD"]
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": NO_CONFIG}
+    except TypeError:
+        return {"status": DOWN}
+    log.debug("got past getting conf")
+    try:
+        start = datetime.now()
+        with Timeout(2):
+            log.debug("entered context manager")
+            log.debug(pformat(conf))
+            connection = connect(
+                database=database, user=user, host=host,
+                port=port, password=password
+            )
+            log.debug("at end of context manager")
+        micro = (datetime.now() - start).microseconds
+        connection.close()
+    except (OperationalError, KeyError, Timeout.Timeout):
+        log.error("Invalid PostgreSQL connection info in settings: %s", conf)
+        return {"status": DOWN}
+    log.debug("got to end of postgres check successfully")
+    return {"status": UP, "response_microseconds": micro}
+
+
+def get_redis_info():
+    """Check Redis connection."""
+    try:
+        url = settings.BROKER_URL
+        _, host, port, _, _, db, _ = parse_redis_url(url)
+    except AttributeError:
+        log.error("No valid Redis connection info found in settings.")
+        return {"status": NO_CONFIG}
+
+    start = datetime.now()
+    try:
+        with Timeout(2):
```

Comment:
Can we make this configurable with a default, or at least a constant at the top of the file to make it easier to change later if 2 seconds proves to be problematic?